### PR TITLE
Add redirects for most of www/www-deploy.yml

### PR DESCRIPTION
**Pull request by manics**
Add redirects for most of https://github.com/ome/prod-playbooks/blob/3027ede41396c49cb8ae4c5dede19a58a718b6e6/www/www-deploy.yml

```python

import os
import re
import yaml

with open('../prod-playbooks/www/www-deploy.yml') as f:
    y = yaml.safe_load(f)

redirects = y[0]['vars']['nginx_proxy_redirect_map']

index_redirect = '---\nredirect_to: {dest}\n---\n'

cleanre = re.compile(r'~/(.+[^/?.*$])[/?.*$]*$')
for r in redirects:
    if r['match'] == 'default' or '(' in r['match']:
        print('❌', r['match'])
        continue

    m = cleanre.match(r['match'])
    if m:
        # src = os.path.split(m.group(1))
        src = m.group(1).replace(r'\.', '.')
        dest = r['dest']
        print('✅', src, '➜', dest)
        os.makedirs(src, exist_ok=True)
        with open(os.path.join(src, 'index.html'), 'w') as f:
            f.write(index_redirect.format(dest=dest))
    else:
        print('❌', r['match'])

```

### CODE REVIEW

**Comment by manics** [comment]
Look like https://github.com/ome/prod-playbooks/blob/3027ede41396c49cb8ae4c5dede19a58a718b6e6/www/www-deploy.yml#L87 is incorrect

File: site/about/development-teams/glencoe-software/index.html [hunk 1]
```diff
@@ -0,0 +1,3 @@
+---
+redirect_to: https://www.glencoesoftware.com/team.html
```
Comment (sbesson) [inline site/about/development-teams/glencoe-software/index.html:2]:
```suggestion
redirect_to: https://www.glencoesoftware.com/about/team/
```

File: info/demo/index.html [hunk 2]
```diff
@@ -0,0 +1,3 @@
+---
+redirect_to: http://help.openmicroscopy.org/demo-server.html
```
Comment (jburel) [inline info/demo/index.html:2]: that page redirects to https://www.openmicroscopy.org/explore/
so it does not really make sense

Comment (sbesson) [reply info/demo/index.html:2]: Would you redirect to `/explore` directly? Can you make a suggestion so that we can move forward with this PR?

**Comment by manics** [comment]
A record of the output:
```
❌ default
✅ omero-blog ➜ http://blog.openmicroscopy.org
✅ site ➜ /
✅ site/news ➜ /announcements
✅ site/about ➜ /about
✅ site/about/who-ome ➜ /teams
✅ site/about/licensing ➜ /licensing
❌ ~/site/about/licensing-attribution(/.*)?$
✅ site/about/ome-contributors ➜ /contributors
✅ site/about/partners ➜ /commercial-partners
✅ site/about/development-teams ➜ /teams
✅ site/about/development-teams/glencoe-software ➜ https://www.glencoesoftware.com/team.html
✅ site/about/publications ➜ /citing-ome
✅ site/about/what-omero/overview ➜ /omero
❌ ~/site/about/(?<link>.*)$
✅ site/products ➜ /products
✅ site/products/omero ➜ /omero
✅ site/products/omero/downloads ➜ /omero/downloads/
✅ site/products/omero/feature-list ➜ /omero/features/
✅ site/products/omero/big-images-support ➜ /omero/view/
✅ site/products/omero/secvuln ➜ /security/advisories/
✅ site/products/ome5/secvuln ➜ /security/advisories/
❌ ~/site/products/omero/secvuln/(?<link>.*[^/])/?$
✅ site/products/bio-formats ➜ /bio-formats
✅ site/products/bio-formats/downloads ➜ /bio-formats/downloads/
✅ site/products/ome-files-cpp ➜ /ome-files
❌ ~/site/products/(?<link>.*)$
✅ forums ➜ https://forum.image.sc/c/data-management
✅ site/community ➜ /support
✅ site/community/mailing-lists ➜ /support
❌ ~/site/events(/.*)?$
✅ site/community/minutes/conference-calls ➜ /on-the-web
✅ site/community/minutes/meetings/12th-annual-users-meeting-2017 ➜ /events/12th-annual-users-meeting-2017.html
✅ site/community/minutes/meetings/11th-annual-users-meeting-2016 ➜ /events/11th-annual-users-meeting-2016.html
✅ site/community/minutes/meetings/10th-annual-users-meeting-june-2015 ➜ /events/10th-annual-users-meeting-june-2015.html
✅ site/community/minutes/meetings/9th-annual-users-meeting-june-2014 ➜ /events/9th-annual-users-meeting-june-2014.html
✅ site/community/jobs ➜ /careers
✅ site/community/scripts ➜ https://docs.openmicroscopy.org/latest/omero/developers/scripts/index.html
❌ ~/site/community/(?<link>.*)$
✅ site/support ➜ /docs
✅ site/support/bio-formats ➜ https://docs.openmicroscopy.org/latest/bio-formats/
❌ ~/site/support/bio-formats/(?<link>.*)$
✅ site/support/bio-formats5 ➜ https://docs.openmicroscopy.org/latest/bio-formats5/
❌ ~/site/support/bio-formats5/(?<link>.*)$
✅ site/support/bio-formats5.3 ➜ https://docs.openmicroscopy.org/latest/bio-formats5.3/
❌ ~/site/support/bio-formats5\.3/(?<link>.*)$
✅ site/support/bio-formats5.4 ➜ https://docs.openmicroscopy.org/latest/bio-formats5.4/
❌ ~/site/support/bio-formats5\.4/(?<link>.*)$
✅ site/support/bio-formats5.5 ➜ https://docs.openmicroscopy.org/latest/bio-formats5.5/
❌ ~/site/support/bio-formats5\.5/(?<link>.*)$
✅ site/support/omero ➜ https://docs.openmicroscopy.org/latest/omero/
❌ ~/site/support/omero/(?<link>.*)$
✅ site/support/omero5 ➜ https://docs.openmicroscopy.org/latest/omero5/
❌ ~/site/support/omero5/(?<link>.*)$
✅ site/support/omero5.0 ➜ https://docs.openmicroscopy.org/latest/omero5.0/
❌ ~/site/support/omero5\.0/(?<link>.*)$
✅ site/support/omero5.1 ➜ https://docs.openmicroscopy.org/latest/omero5.1/
❌ ~/site/support/omero5\.1/(?<link>.*)$
✅ site/support/omero5.2 ➜ https://docs.openmicroscopy.org/latest/omero5.2/
❌ ~/site/support/omero5\.2/(?<link>.*)$
✅ site/support/omero5.3 ➜ https://docs.openmicroscopy.org/latest/omero5.3/
❌ ~/site/support/omero5\.3/(?<link>.*)$
✅ site/support/ome-model ➜ https://docs.openmicroscopy.org/latest/ome-model/
❌ ~/site/support/ome-model/(?<link>.*)$
✅ site/support/file-formats ➜ https://docs.openmicroscopy.org/latest/ome-model/
✅ site/support/file-formats/schemas/specifications/compliant-file-specification ➜ https://docs.openmicroscopy.org/latest/ome-model/specifications/
✅ site/support/ome-tiff ➜ https://docs.openmicroscopy.org/latest/ome-model/ome-tiff/
✅ site/support/ome-files-cpp ➜ https://docs.openmicroscopy.org/latest/ome-files-cpp/
❌ ~/site/support/ome-files-cpp/(?<link>.*)$
✅ site/support/contributing ➜ https://docs.openmicroscopy.org/contributing/
❌ ~/site/support/contributing/(?<link>.*)$
❌ ~/site/support/previous(/.*)?$
❌ ~/site/support/ome-artwork(/.*)?$
✅ BIO-FORMATS ➜ /bio-formats
❌ ~/BIO-FORMATS/(?<link>.*)$
✅ OME-FILES ➜ /ome-files
❌ ~/OME-FILES/(?<link>.*)$
✅ OMERO ➜ /omero
❌ ~/OMERO/(?<link>.*)$
✅ info/vulnerabilities ➜ /security/advisories/
❌ ~/info/vulnerabilities/(?<link>.*[^/])/?$
✅ info/omero ➜ /omero
✅ info/OMERO.insight ➜ https://docs.openmicroscopy.org/latest/omero/users/index.html
✅ info/OMERO.importer ➜ https://docs.openmicroscopy.org/latest/omero/users/index.html
✅ info/OMERO.editor ➜ https://docs.openmicroscopy.org/latest/omero/users/index.html
✅ info/OMERO.web ➜ https://docs.openmicroscopy.org/latest/omero/users/index.html
✅ info/OMERO.server ➜ https://docs.openmicroscopy.org/latest/omero/users/index.html
✅ info/permissions ➜ https://docs.openmicroscopy.org/latest/omero/sysadmins/server-permissions.html
✅ info/demo ➜ http://help.openmicroscopy.org/demo-server.html
✅ info/cls ➜ /omero/downloads/
✅ info/download ➜ /omero/downloads/
✅ info/downloads ➜ /omero/downloads/
✅ info/lists ➜ http://lists.openmicroscopy.org.uk/mailman/listinfo/
✅ info/videos ➜ https://www.youtube.com/channel/UCyySB9ZzNi8aBGYqcxSrauQ
✅ info/attribution ➜ /licensing/
✅ info/downgrade ➜ https://docs.openmicroscopy.org/latest/omero/developers/Model/XsltTransformations.html
✅ info/flimfit ➜ http://flimfit.org
✅ info/scripts ➜ https://docs.openmicroscopy.org/latest/omero/developers/scripts/index.html
❌ ~/info/bio-formats(/.*)?$
✅ info/slidebook ➜ https://www.intelligent-imaging.com/technical-answers
❌ ~/info/(.*)?$
```

The following were manually reverted (directories already exist and OS X is case-insensitive):
```
✅ BIO-FORMATS ➜ /bio-formats
✅ OME-FILES ➜ /ome-files
✅ OMERO ➜ /omero
```

File: info/demo/index.html (hunk 2 above)
Comment (manics) [inline info/demo/index.html:2]:
```suggestion
redirect_to: https://www.openmicroscopy.org/explore/
```

**Comment by sbesson** [comment]
Looking briefly at the latest failures in https://travis-ci.org/github/ome/www.openmicroscopy.org/builds/735673501, I think they correspond to the deep redirects which are not handled by the current PR incl.

```
/community/**
/qa2/**
/site/support/**/*
/site/security/advisories/**
```

How do you see proceeding with these beyond this PR? Unfortunately, `jekyll-redirect-from` does not handle wildcard as far as I can tell.

**Comment by manics** [comment]
Most of the `/site/support/*` links go to something under `https://docs.openmicroscopy.org/` so although we could add redirects they'd still be broken at the moment. If you think it's helpful I can attempt that in this PR but there's no way of testing whether it's correct since the destination is down.

I can take a look at the other failures, some of them are double redirects so easy to fix.

**Comment by sbesson** [comment]
`/site/support/**` is tricky. I agree we could fix the 10ish references within the existing announcement pages but that is only addressing the symptoms. In general, how do you envision the strategy for handling these redirects at the Jekyll/GitHub pages level?

**Comment by manics** [comment]
Two options:
- Add in the deep redirect
- Redirect to an Nginx that we control, that does a further redirect to the real target

**Comment by manics** [comment]
I've fixed most of the links, and set the linkcheck to ignore:
- `https://www.openmicroscopy.org/site/support/bio-formats5`
- `https://www.openmicroscopy.org/community`
- `https://www.openmicroscopy.org/qa2`
There are 2 remainining failing links which don't make sense to me since the pages exist:
```
- /site/2016/05/10/ome-development-status.html

  *  internally linking to /events/11th-annual-users-meeting-2016/, which does not exist (line 119)

     <a href="/events/11th-annual-users-meeting-2016/">OME Users meeting</a>

- /site/2017/02/02/ome-users-meeting.html

  *  internally linking to /events/12th-annual-users-meeting-2017/, which does not exist (line 121)

     <a href="/events/12th-annual-users-meeting-2017/">meeting 2017</a>
```

File: _posts/2016-05-10-ome-development-status.md [hunk 3]
```diff
@@ -3,7 +3,7 @@ layout: post
 title: OME Development Status
 intro-blurb: Status of development and focus shifts for OMERO, Bio-Formats, CI and documentation
 ---
-With the recent release of OMERO 5.2.3 and Bio-Formats 5.1.10, the OME team is now focusing on the preparation of the workshops for our annual [OME Users meeting](https://www.openmicroscopy.org/site/community/minutes/meetings/11th-annual-users-meeting-2016).
+With the recent release of OMERO 5.2.3 and Bio-Formats 5.1.10, the OME team is now focusing on the preparation of the workshops for our annual [OME Users meeting]({{ site.baseurl }}/events/11th-annual-users-meeting-2016/).
```
Comment (sbesson) [inline _posts/2016-05-10-ome-development-status.md:6]:
```suggestion
With the recent release of OMERO 5.2.3 and Bio-Formats 5.1.10, the OME team is now focusing on the preparation of the workshops for our annual [OME Users meeting]({{ site.baseurl }}/events/11th-annual-users-meeting-2016.html).
```

File: _posts/2017-02-02-ome-users-meeting.md [hunk 4]
```diff
@@ -5,7 +5,7 @@ intro-blurb: The 12th Annual OME Users Meeting will be in Dundee on the 31st May
 ---
 Currently, we aim to start on the morning of May 31 with community lightning talks followed by an afternoon of longer presentations. On June 1, we’ll have a day of workshops and discussions. We’d love feedback on last year’s UnMeeting. If you want to plan on a day of coding, maybe we can do that on June 2. If you’re interested, let us know.
 
-For further details, see [meeting 2017](https://www.openmicroscopy.org/site/events/ome-users-meeting-2017)
+For further details, see [meeting 2017]({{ site.baseurl }}/events/12th-annual-users-meeting-2017/)
```
Comment (sbesson) [inline _posts/2017-02-02-ome-users-meeting.md:8]:
```suggestion
For further details, see [meeting 2017]({{ site.baseurl }}/events/12th-annual-users-meeting-2017.html)
```

**Comment by manics** [comment]
To get the `https://www.openmicroscopy.org/community` and `https://www.openmicroscopy.org/qa2` links passing I think we'll need to redirect to a host on a different domain that we manage.

File: about/index.html [hunk 5]
```diff
@@ -203,7 +203,7 @@ <h5 class="subheader">The members of the OME consortium are as follows:</h5>
         <hr>
         <div class="row medium-up-1 large-up-1 text-center consortium-list">
             {% for member in page.consortium %}
-            <div class="large-expand columns"><a target="_blank" href="{{ member.member_link }}">{{ member.member_name }}</a> <span class="institution">{{ member.institution }}</span></div>
+            <div class="large-expand columns"><a target="_blank" href="{{ member.member_link | replace: '!SITE_BASEURL!', site.baseurl }}">{{ member.member_name }}</a> <span class="institution">{{ member.institution }}</span></div>
```
Comment (sbesson) [inline about/index.html:206]: This and the same below is to ensure that a breaking change would be flagged during a PR review?

**Review by sbesson — APPROVED** [review]
Overall, the update of the historical links makes sense and the builds are back to a passing state.

By order of priority I think the remaining sub-directories missing from the GH deployment are:

- `/Schemas` - should be handled by https://github.com/ome/www.openmicroscopy.org/pull/420
- `/qa2/**` https://www-legacy.openmicroscopy.org/qa2/ should now serve the QA application if we wanted to test the proxying solution
- `/community/**` assuming we can spin up a static website with the static copy of the old phpbb forum
- some of the deep wildcard redirect - in some instances we can probably convert these to a series of explicit redirect but I assume a subset will still need some proxying solution as above

Overall, happy for this to merge this immediately as it fixes Travis CI, unblocks other improvements and helps us getting this website in a state close the the one deployed by https://github.com/ome/prod-playbooks/blob/master/www/www-deploy.yml.

As discussed this morning, we probably still need to discuss whether this is sufficient or switch the DNS back to ome-www and make progress on the remaining components.

**Comment by manics** [comment]
Screen Shot 2020-10-15 at 17 21 32

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [WEBSITE-push#110](https://merge-ci.openmicroscopy.org/jenkins/job/WEBSITE-push/110/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/WEBSITE-push/110/consoleText) for more details.
Possible conflicts:
  - Upstream changes
    - .travis.yml

--conflicts

**Comment by jburel** [comment]
Travis file is now gone leading to a conflict

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [WEBSITE-push#83](https://latest-ci.openmicroscopy.org/jenkins/job/WEBSITE-push/83/). See the [console output](https://latest-ci.openmicroscopy.org/jenkins/job/WEBSITE-push/83/consoleText) for more details.
Possible conflicts:
  - Upstream changes
    - .travis.yml

--conflicts

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [WEBSITE-push#83](https://latest-ci.openmicroscopy.org/jenkins/job/WEBSITE-push/83/). See the [console output](https://latest-ci.openmicroscopy.org/jenkins/job/WEBSITE-push/83/consoleText) for more details.
Possible conflicts:
  - Upstream changes
    - .travis.yml

--conflicts

**Comment by pwalczysko** [comment]
An approved PR, but has a conflict. Hence, bypassing it for this release.

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [WEBSITE-push#135](https://merge-ci.openmicroscopy.org/jenkins/job/WEBSITE-push/135/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/WEBSITE-push/135/consoleText) for more details.
Possible conflicts:
  - Upstream changes
    - .travis.yml

--conflicts

**Comment by sbesson** [comment]
Fixing the conflicts will require

**Pull request closed without merging**